### PR TITLE
Delete unused `Violation::explanation` method

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

Deletes the unused `Violation::explanation` method

## Test Plan

`cargo build`

### CODE REVIEW

File: crates/ruff_diagnostics/src/violation.rs [hunk 1]
```diff
@@ -25,11 +25,6 @@ pub trait Violation: Debug + PartialEq + Eq {
     /// The message used to describe the violation.
     fn message(&self) -> String;
 
-    /// The explanation used in documentation and elsewhere.
-    fn explanation() -> Option<&'static str> {
-        None
-    }
-
     // TODO(micha): Move `fix_title` to `Fix`, add new `advice` method that is shown as an advice.
```
Comment (MichaReiser) [inline crates/ruff_diagnostics/src/violation.rs:28]: Let's see how many years it takes before I have time to work on this :laughing:

**Review by AlexWaygood — APPROVED** [review]

**Comment by dhruvmanila** [comment]
Nice! Does this mean that not adding a documentation is a compiler error?

**Comment by MichaReiser** [comment]
> Nice! Does this mean that not adding a documentation is a compiler error?

That I don't know. I only removed the `explanation` method. I assume that the explanation was never used during runtime. `cargo dev generate-all` must store it somewhere else.

**Comment by dhruvmanila** [comment]
Yeah, the `violation` macro generates the `explanation` method only if there's a doc comment on the struct. It does give me a compiler error if there's no such method because there's no default implementation now:

```
rustc: function or associated item `explanation` not found for this struct
     crates/ruff_linter/src/codes.rs(65:1): original diagnostic [E0599]
```

**Comment by MichaReiser** [comment]
I guess that's a nice added benefit :D 

Does that mean that we never called the method using the trait directly.

**Comment by dhruvmanila** [comment]
> Does that mean that we never called the method using the trait directly.

Yeah, I don't think so. The macro must be directly calling it like `rule.explanation()` 🤷‍♂️

**Pull request merged**